src/haddock/modules/analysis/caprieval/__init__.py

```diff
@@ -23,6 +27,32 @@ def confirm_installation(cls):
         """Confirm if contact executable is compiled."""
         return
 
+    def _rearrange_output(self, output_name, path, ncores):
+        """Combine different capri outputs in a single file."""
+        output_fname = Path(path, output_name)
+        log.info(f"rearranging output files into {output_fname}")
+        keyword = output_name.split(".")[0]
+        split_dict = {
+            "capri_ss": "model-cluster-ranking",
+            "capri_clt": "caprieval_rank"
+            }
+        if keyword not in split_dict.keys():
+            raise Exception(f'Keyword {keyword} does not exist.')
+        # Combine files
+        with open(output_fname, 'w') as out_file:
+            for core in range(ncores):
+                tmp_file = Path(path, keyword + "_" + str(core) + ".tsv")
+                with open(tmp_file) as infile:
+                    if core == 0:
+                        out_file.write(infile.read().rstrip("\n"))
+                    else:
+                        kw = split_dict[keyword]
+                        out_file.write(infile.read().split(kw)[1].rstrip("\n"))
```

Comment:
rstripping the linesep character is necessary to avoid having holes in the reconstructed file, but I agree that we shouldn't be doing that on the last file. I will adjust it accordingly.

Reply:
You can do the `rstrip`. Simply, add a `+ os.linesep` after it as I wrote. I think that should solve the problem but I haven't tested though.

Reply:
Yes, exactly, I did that and I'm about to test it. should be fine

Reply:
is fine. I tested it